lib/service/database.js

```diff
@@ -294,7 +298,8 @@ DatabaseService.prototype.queueAnalysisOperations = function(analysis, callback)
                     id: asyncQueryId(analysis, nodeToUpdate),
                     query: transactionQuery([
                         deleteFromCacheTableQuery(targetTableName),
-                        populateCacheTableQuery(targetTableName, nodeToUpdate.sql())
+                        populateCacheTableQuery(targetTableName, nodeToUpdate.sql()),
+                        checkCacheTableQuery(targetTableName)
```

Comment:
Now that I think of it... if we only create cached tables when calling `populateCacheTableQuery` and we check after each call to it, we're already failing as soon as we can. (note that we check space allocated to cache tables) 😁